### DISCUSSION TITLE
Switching browser tabs kills Supabase

### DISCUSSION THREAD

**Discussion by wes-prompt** [supabase/supabase#17612] [Questions]
When switching to a different browser tab away from my NextJS application and coming back, Supabase becomes unresponsive and returns no data, errors or status. Below is a test page I've made to reproduce this. Simply click the button. Navigate to a new tab. Come back. Click the button again and nothing except "before supabase" will be written to the console. Anyone know ways around this problem?

'use client'

import {createClientComponentClient} from '@supabase/auth-helpers-nextjs'

export default function Test() {
	const supabase = createClientComponentClient()

	const callSupabase = async () => {
		console.log('before supabase')

		const {data, error, status} = await supabase
		.from('accounts')
		.select()

		console.log(data, error, status)
	}

	return (
		<button onClick={callSupabase}>Call Supabase</button>
	)
}

**Comment by izhan** [comment]
We are seeing this issue as well, version v2.39.1 for supabase-js

Possibly related to https://github.com/supabase/gotrue-js/issues/762

**Comment by ScreamZ** [comment]
See my answer in https://github.com/nuxt-modules/supabase/issues/273#issuecomment-2051932773

**Comment by edarioq** [comment]
I am still experiencing this issue. What is the fix ? I've tried installing `@supabase/gotrue-js` but nothing.

**Reply by GaryAustin1** [reply]
You probably need to provide more specifics as in general this is not an issue.

**Comment by wes-prompt** [comment]
My problem was including “supabase” as a dependency in the useEffects. Once I removed the dependency the issue went away.

**Comment by kerryegeler** [comment]
I'm still facing this issue in my software.

When a user is logged in, then switches to another tab, then back, the supabase data will not load until the user manually refreshes the browser.

I've tried all the fixes here with no luck. Can anyone help me figure out what it might be?

**Reply by docaohuynhcse** [reply]
I facing exactly the same here. Can't find anything to fix this

**Reply by R3CK0** [reply]
Yeah have the same issue, spent 4 hours trying to find a fix today nothing seems to work

**Reply by j4w8n** [reply]
I'm not having any issues in my demo sveltekit app. You might need to share more details.

**Comment by docaohuynh** [comment]
I have the same issue in angular but now it works

[repo work](https://github.com/docaohuynh/angular-supabase-auth)
